### PR TITLE
GRPCv2 Add endpoints needed for delegation

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Add missing grpcv2 endpoints that are needed for delegation.

## Changes

Added 
    - `getBlockChainParameters`
    - `getPoolInfo`
    - `getPassiveDelegationInfo`
    - `getTokenomicsInfo`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/nodejs/src/typeTranslation.ts [hunk 1]
```diff
@@ -299,3 +418,188 @@ export function cryptographicParameters(
...
+    }
+}
+
+export function bakerPoolInfo(info: v2.PoolInfoResponse): v1.BakerPoolStatus {
+    return {
+        poolType: PoolStatusType.BakerPool,
+        bakerId: unwrap(info.baker?.value),
+        bakerAddress: AccountAddress.fromBytes(
+            Buffer.from(unwrap(info.address?.value))
+        ).address,
```
Comment (rasmus-kirk) [inline packages/nodejs/src/typeTranslation.ts:538]: Use `trAccountAddress`

File: packages/nodejs/src/typeTranslation.ts [hunk 2]
```diff
@@ -299,3 +418,188 @@ export function cryptographicParameters(
         genesisString: cp.genesisString,
     };
 }
+
+export function blockChainParameters(
+    params: v2.ChainParameters
+): v1.ChainParameters {
```
Comment (rasmus-kirk) [inline packages/nodejs/src/typeTranslation.ts:424]: Create a take in `chainParameters` as argument and define `const cp = chainParameters.parameters`

File: packages/nodejs/src/typeTranslation.ts [hunk 3]
```diff
@@ -235,6 +244,116 @@ function transBaker(
...
+        metadataUrl: info.url,
+        commissionRates: transCommissionRates(info.commissionRates),
+    };
+}
+
+function transPaydayStatus(
+    status: v2.PoolCurrentPaydayInfo | undefined
+): v1.CurrentPaydayBakerPoolStatus | undefined {
+    if (!status) {
+        return undefined;
```
Comment (rasmus-kirk) [inline packages/nodejs/src/typeTranslation.ts:344]: Is there a reason for not just unwrapping status? Can it be undefined in v2?

File: packages/nodejs/src/typeTranslation.ts (hunk 2 above)
Comment (rasmus-kirk) [reply packages/nodejs/src/typeTranslation.ts:424]: Or just define `const v1 = params.parameters.v1` and `const v0 = params.parameters.v0`

File: packages/nodejs/src/typeTranslation.ts [hunk 4]
```diff
@@ -235,6 +244,116 @@ function transBaker(
...
+    change: v2.PoolPendingChange | undefined
+): v1.BakerPoolPendingChange {
+    switch (change?.change?.oneofKind) {
+        case 'reduce': {
+            return {
+                pendingChangeType:
+                    BakerPoolPendingChangeType.ReduceBakerCapital,
+                effectiveTime: new Date(
+                    Number(unwrap(change.change.reduce.effectiveTime?.value))
+                ),
```
Comment (rasmus-kirk) [inline packages/nodejs/src/typeTranslation.ts:311]: `transTimestamp()`? Note that `transTimestamp()` multiplies with 1000 but should we also do this here?

File: packages/nodejs/src/typeTranslation.ts [hunk 5]
```diff
@@ -235,6 +244,116 @@ function transBaker(
...
+                    change.change.reduce.reducedEquityCapital?.value
+                ),
+            };
+        }
+        case 'remove': {
+            return {
+                pendingChangeType: BakerPoolPendingChangeType.RemovePool,
+                effectiveTime: new Date(
+                    Number(unwrap(change.change.remove.effectiveTime?.value))
+                ),
```
Comment (rasmus-kirk) [inline packages/nodejs/src/typeTranslation.ts:322]: Same `transTimestamp()` comment

File: packages/nodejs/src/clientV2.ts [hunk 6]
```diff
@@ -120,4 +120,66 @@ export default class ConcordiumNodeClient {
...
+    }
+
+    /**
+     * Retrieves information on the baker pool of the given bakerId.
+     * @param blockHash the block hash of the block to get the information from.
+     * @param bakerId the ID of the baker to get the status for.
+     * @returns The status of the corresponding baker pool.
+     */
+    async getPoolInfo(
+        bakerId: bigint,
```
Comment (rasmus-kirk) [inline packages/nodejs/src/clientV2.ts:146]:
```suggestion
        bakerId: v1.BakerId,
```

File: packages/common/src/types.ts [hunk 7]
```diff
@@ -448,7 +448,7 @@ export interface GasRewards {
     chainUpdate: number;
 }
 
-interface RewardParametersCommon {
+export interface RewardParametersCommon {
     transactionFeeDistribution: TransactionFeeDistribution;
     gASRewards: GasRewards;
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/common/src/types.ts:453]: I think the "desired" capitalization was GASRewards, but then to keep lower camelCase, the g was made small. But I'm not sure, this capitalization is from the node.

File: packages/nodejs/test/clientV2.test.ts [hunk 8]
```diff
@@ -288,3 +288,54 @@ test('accountInfo implementations is the same', async () => {
...
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const blockSummary: any = await clientV1.getBlockSummary(oldBlockHash);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const chainParameters: any = await clientV2.getBlockChainParameters(
+        oldBlockHash
+    );
+
+    // TODO Fix the foundation account change
+    blockSummary.updates.chainParameters.foundationAccountIndex = undefined;
+    chainParameters.foundationAccount = undefined;
```
Comment (rasmus-kirk) [inline packages/nodejs/test/clientV2.test.ts:318]: Lookup foundationAccountIndex

File: packages/common/src/types.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (rasmus-kirk) [reply packages/common/src/types.ts:453]: ¯\\_(ツ)_/¯

File: packages/nodejs/src/typeTranslation.ts (hunk 3 above)
Comment (shjortConcordium) [reply packages/nodejs/src/typeTranslation.ts:344]: Yeah:
```
* @generated from protobuf field: optional concordium.v2.PoolCurrentPaydayInfo current_payday_info = 8;
```

File: packages/nodejs/src/typeTranslation.ts (hunk 4 above)
Comment (shjortConcordium) [reply packages/nodejs/src/typeTranslation.ts:311]: I've changed it to use transTimestamp, but it should not multiply with 1000. (But checking the types/comments, it should never do that in the used cases, the timestamp should be in milliseconds and the date constructor expects a number argument to be in milliseconds)

Comment (shjortConcordium) [reply packages/nodejs/src/typeTranslation.ts]: (Okay I've broken a test now. I'll look at it tomorrow)

File: packages/common/src/GRPCTypeTranslation.ts [hunk 9]
```diff
@@ -293,8 +299,111 @@ function trBaker(baker: v2.AccountStakingInfo_Baker): v1.AccountBakerDetails {
     };
 }
 
+function trAccountAddressToString(
+    accountAddress: v2.AccountAddress | undefined
+): string {
```
Comment (rasmus-kirk) [inline packages/common/src/GRPCTypeTranslation.ts:304]: This function is just `unwrapToBase58()`

File: packages/nodejs/test/clientV2.test.ts [hunk 10]
```diff
@@ -234,6 +230,100 @@ test.each([clientV2, clientWeb])(
...
+        chainParameters.foundationAccount = undefined;
+
+        expect(blockSummary.updates.chainParameters).toEqual(chainParameters);
+    }
+);
+
+test.each([clientV2, clientWeb])(
+    'getPoolInfo corresponds to getPoolStatus with a bakerId',
+    async (client) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rasmus-kirk) [inline packages/nodejs/test/clientV2.test.ts:284]: Why this any? And if they return the same object, why not keep the old function name?

Comment (rasmus-kirk) [reply packages/nodejs/test/clientV2.test.ts]: It's fine

**Comment by rasmus-kirk** [comment]
Resolve and I'll approve

**Comment by rasmus-kirk** [comment]
Should work now, look it over too and just merge if it looks good to you

@shjortConcordium

**Review by rasmus-kirk — APPROVED** [review]

File: packages/nodejs/test/clientV2.test.ts [hunk 11]
```diff
@@ -212,6 +212,89 @@ test.each([clientV2, clientWeb])(
     }
 );
 
+test.each([clientV2, clientWeb])(
+    'getChainParameters corresponds to GetBlockSummary subset',
+    async (client) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rasmus-kirk) [inline packages/nodejs/test/clientV2.test.ts:218]: Are these necessary

File: packages/nodejs/test/clientV2.test.ts [hunk 12]
```diff
@@ -212,6 +212,89 @@ test.each([clientV2, clientWeb])(
...
+        expect(blockSummary.updates.chainParameters).toEqual(chainParameters);
+    }
+);
+
+test.each([clientV2, clientWeb])(
+    'getChainParameters corresponds to GetBlockSummary subset on protocol level < 4',
+    async (client) => {
+        const oldBlockHash =
+            'ed2507c4d05108038741e87757ab1c3acdeeb3327027cd2972666807c9c4a20d';
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rasmus-kirk) [inline packages/nodejs/test/clientV2.test.ts:243]: And these

**Pull request merged**